frontend/src/components/Quiz/Category.jsx

```
@@ -0,0 +1,99 @@
+        console.error('Error fetching categories:', err);
+        setLoading(false);
+      });
+  }, []);
+
+  const handleCategorySelect = (categoryId) => {
+    navigate(`/quiz/${categoryId}`);
+  };
+
+  if (loading) return (
+    <div className="flex justify-center items-center h-64">
+      <div className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-blue-500"></div>
```

Biome (1.9.4)

[error] JSX elements without children should be marked as self-closing. In JSX, it is valid for any element to be self-closing.

Unsafe fix: Use a SelfClosingElement instead


(lint/style/useSelfClosingElements)